packages/puppeteer-core/src/bidi/Browser.ts

```diff
@@ -70,47 +72,28 @@ export class BidiBrowser extends Browser {
   ];
 
   static async create(opts: BidiBrowserOptions): Promise<BidiBrowser> {
-    let browserName = '';
-    let browserVersion = '';
-
-    // TODO: await until the connection is established.
-    try {
-      const {result} = await opts.connection.send('session.new', {
-        capabilities: {
-          alwaysMatch: {
-            acceptInsecureCerts: opts.ignoreHTTPSErrors,
-          },
-        },
-      });
-      browserName = result.capabilities.browserName ?? '';
-      browserVersion = result.capabilities.browserVersion ?? '';
-    } catch (err) {
-      // Chrome does not support session.new.
-      debugError(err);
-    }
-
-    await opts.connection.send('session.subscribe', {
-      events: browserName.toLocaleLowerCase().includes('firefox')
-        ? BidiBrowser.subscribeModules
-        : [...BidiBrowser.subscribeModules, ...BidiBrowser.subscribeCdpEvents],
+    const session = await Session.from(opts.connection, {
+      alwaysMatch: {
+        acceptInsecureCerts: opts.ignoreHTTPSErrors,
+        webSocketUrl: true,
+      },
     });
 
-    const browser = new BidiBrowser({
-      ...opts,
-      browserName,
-      browserVersion,
-    });
+    await session.subscribe(
+      session.capabilities.browserName.toLocaleLowerCase().includes('firefox')
+        ? BidiBrowser.subscribeModules
+        : [...BidiBrowser.subscribeModules, ...BidiBrowser.subscribeCdpEvents]
+    );
 
+    const browser = new BidiBrowser(session.browser, opts);
     await browser.#getTree();
-
+    browser.#initialize();
```

Comment:
I think this should happen before browser.#getTree() I believe

Reply:
You're right.